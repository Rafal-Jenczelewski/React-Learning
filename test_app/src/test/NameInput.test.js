import React from 'react';
import ConnectedNameInput, {NameInput} from '../main/components/NameInput';
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Provider} from 'react-redux';
import {changeName} from '../main/actions/index';

Enzyme.configure({adapter: new Adapter()});

describe('Dumb NameInput', () => {
    let wrapper;

    it('renders without crashing', () => {
        wrapper = shallow(<NameInput/>);
        expect(wrapper.length).toBe(1);
    });

    it('matches snapshot', () => {
        wrapper = renderer.create(<NameInput/>);

        let tree = wrapper.toJSON();
        expect(tree).toMatchSnapshot('Name Input snapshot');
    });

    describe('When input changes', () => {
        it('calls handler', () => {
            let spy = jest.fn();

            NameInput.prototype.changeHandler = spy;
            wrapper = mount(<NameInput/>);
            wrapper.update();
            wrapper.find('input').simulate('change', {});

            expect(spy).toBeCalled();
        });
    });
});

describe('Connected NameInput', () => {
    const initialState = {name: 'World'};
    const mockStore = configureStore();
    let store, wrapper;

    beforeEach(() => {
        store = mockStore(initialState);
    });

    it('renders without crashing', () => {
        wrapper = shallow(<ConnectedNameInput store={store}/>);
        expect(wrapper.length).toBe(1);
    });

    it('matches snapshot', () => {
        wrapper = renderer.create(<ConnectedNameInput store={store}/>)

        let tree = wrapper.toJSON();
        expect(tree).toMatchSnapshot("ConnectedNameInput")
    });
});