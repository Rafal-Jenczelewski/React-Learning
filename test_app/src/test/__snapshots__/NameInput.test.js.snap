// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedNameInput 1`] = `
<div
  className="csfWidgets"
>
  <div
    className="header-height"
  >
    <div
      className="csfWidgets label"
      id={null}
    >
      <label
        className="top"
        htmlFor="MakeUnique"
        onClick={[Function]}
      >
        My name is
      </label>
    </div>
  </div>
  <div
    className="has-outline textinput "
  >
    <div
      className="testTextInput"
    >
      <input
        aria-label="textinput"
        aria-multiline={false}
        aria-readonly={false}
        className="text"
        disabled={false}
        id="MakeUnique"
        name=""
        onBlur={[Function]}
        onChange={[Function]}
        onCut={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        placeholder=""
        readOnly={false}
        required={false}
        spellCheck={true}
        style={null}
        tabIndex={0}
        type="text"
        value="World"
      />
    </div>
    <div
      className="footer-height"
    />
  </div>
</div>
`;

exports[`Name Input snapshot 1`] = `
<div
  className="csfWidgets"
>
  <div
    className="header-height"
  >
    <div
      className="csfWidgets label"
      id={null}
    >
      <label
        className="top"
        htmlFor="MakeUnique"
        onClick={[Function]}
      >
        My name is
      </label>
    </div>
  </div>
  <div
    className="has-outline textinput "
  >
    <div
      className="testTextInput"
    >
      <input
        aria-label="textinput"
        aria-multiline={false}
        aria-readonly={false}
        className="text"
        disabled={false}
        id="MakeUnique"
        name=""
        onBlur={[Function]}
        onChange={[Function]}
        onCut={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        placeholder=""
        readOnly={false}
        required={false}
        spellCheck={true}
        style={null}
        tabIndex={0}
        type="text"
        value=""
      />
    </div>
    <div
      className="footer-height"
    />
  </div>
</div>
`;
